components/collector/src/base_collectors/collector.py

```diff
@@ -132,6 +132,7 @@ async def collect_metric(
         metric_collector = metric_collector_class(session, metric, self.data_model)
         if measurement := await metric_collector.collect():
             measurement.metric_uuid = metric_uuid
+            measurement.report_uuid = metric.get("report_uuid", "")
```

Comment:
Why the `get("report_uuid)`? The collector gets the metrics from the Internal endpoint `get_metrics` so the metrics are guaranteed to have a report_uuid attribute?

Reply:
Unless this update was for some reason only applied to the collector and not to the server.

Also, there are a lot of (unit)tests that will fail because there is no report_uuid is not present in the mocks. Requiring the report_uuid as a mandatory property would add complexity to all those tests without actually improving their value. So using get() improves the testability of the code.

All in all, using get() felt like a more fail-safe solution, even though I do agree that technically the report_uuid should always be filled.

Reply:
Changing the production code to cater for the unit tests is not a good argument I think. And I think in this case the get() communicates the wrong message to the reader of the code: it suggests the report_uuid is optional while in fact it is not. The code should crash, violently and loudly, if there is no report_uuid, not swallow the issue and then fail somewhere else in the code later on, making it hard to debug.

And I don't agree it adds complexity to the tests. It adds duplication, yes, but that can easily be solved by refactoring the tests.